code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeExtension.java

```diff
@@ -61,11 +62,21 @@ public boolean doWork(final Event event) {
 
     // Map containing the update event IDs (and corresponding requested scopes) for Edge events that
     // haven't yet received an Edge completion response.
-    private Map<String, List<DecisionScope>> updateRequestEventIdsInProgress = new HashMap<>();
+    // Updating the map type from Hashmap to ConcurrentHashMap to avoid the race condition to
```

Comment:
nit: Can you make this comment more concise without mentioning the history?
// Concurrent Map containing the update event IDs (and corresponding requested scopes) for Edge events that haven't yet received an Edge completion response. This is accessed from multiple threads.

Reply:
Updated the comments accordingly @praveek